src/inference/src/ie_core.cpp

```diff
@@ -152,12 +152,14 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
                 _cacheConfig._cacheDir = it->second;
                 if (!it->second.empty()) {
                     FileUtils::createDirectoryRecursive(it->second);
-                    _cacheConfig._cacheManager = std::make_shared<ie::FileStorageCacheManager>(std::move(it->second));
+                    auto cacheDir = it->second;
+                    _cacheConfig._cacheManager = std::make_shared<ie::FileStorageCacheManager>(std::move(cacheDir));
                 } else {
                     _cacheConfig._cacheManager = nullptr;
                 }
 
-                config.erase(it);
+                // WA for CPU plugin. Can be uncommented after implementing snippet serialization.
+                // config.erase(it);
```

Comment:
adding @nosovmik to confirm that we don't break anything with this change

Reply:
It seems the template plugin throws exception.

Reply:
reverted